### PR TITLE
Improve reusable `lint` and `test` workflows

### PR DESCRIPTION
**Pull request by ybiquitous**
> Which issue, if any, is this issue related to?

None.

> Is there anything in the PR that needs further explanation?

This PR includes the followings:

- restrict `permissions`
- run only tests via `npm test`
  - If `npm run pretest` run lint, it should be useful that `npm test` runs only tests without linting

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -45,4 +48,4 @@ jobs:
         run: npm ci
 
       - name: Test
-        run: npm test
+        run: npm test --ignore-scripts
```
Comment (ybiquitous) [inline .github/workflows/test.yml:51]: `npm test --ignore-scripts` works expectedly:

https://github.com/stylelint/stylelint-config-standard/actions/runs/3165150018/jobs/5153954524#step:6:1

**Comment by ybiquitous** [comment]
If this PR is merged, I'll apply the change like https://github.com/stylelint/stylelint-config-standard/pull/262 to other repositories.

**Review by jeddy3 — APPROVED** [review]
LGTM, thank you!

Consistent lint & test behaviour across our repos will be good!

**Comment by ybiquitous** [comment]
> I'll apply the change like [stylelint/stylelint-config-standard#262](https://github.com/stylelint/stylelint-config-standard/pull/262) to other repositories.

Done!

**Comment by ybiquitous** [comment]
All PRs merged. Thanks for your review, @ntwb! 😄

**Pull request merged**